src/daemon/daemon.cpp

```diff
@@ -3942,6 +3988,12 @@ error_string mp::Daemon::async_wait_for_ssh_and_start_mounts_for(
             return fmt::to_string(errors);
         }
         const auto vm = it->second;
+
+        if (vm->current_state() == VirtualMachine::State::unavailable)
```

Comment:
Could you explain the rationale for this check here, for my benefit please? Is it just trying to reduce the chances of AZ and instance state going out of sync due to races?